### ISSUE TITLE
version 0.9.1: [string "Profiles.lua"]:122: attempt to perform arithmetic on a string value--note that 0.9.2 is out now to fix this

### ISSUE THREAD

**Issue by WaellerKlaus** [rsjaffe/MIDI2LR#136]
Many thanks for you very fast implementation of profile changes and the very helpfull concept of the assigment of profiles to develop panels and LR-modules.
The above error message appears always immediately when starting MIDI2LR (release 0.9.1) on Mac OS X 10.10.5 (14F1021) using Lightroom 6.3 [ 1052427 ]. Then the last profile which was loaded manually works just fine. However noch change of profiles is working, neither as assigned to develop panels, nor assigned to control buttons through profile load command, except for manually loading them via the app's UI. Hope you have a chance to look into this. Let me know if you need further information or things I can check further.

**Comment by krakozavr** [comment]
just made the screenshot about this issue :)
![error-midi2lr](https://cloud.githubusercontent.com/assets/16202038/12247036/0a970a52-b880-11e5-985d-8cf7173ef02c.png)

**Comment by rsjaffe** [comment]
Gosh. A routine that worked fine in one module is crashing in another.

Please try this fix:

there are two places in Profiles.lua that have the following code
```lua
      -- refresh MIDI controller since mapping has changed
      for _,param in ipairs(Parameters.Order) do
        local min,max = Limits.GetMinMax(param)
        local midivalue = (LrDevelopController.getValue(param)-min)/(max-min) * MIDI2LR.CONTROL_MAX
        MIDI2LR.SERVER:send(string.format('%s %g\n', param, midivalue))
      end
```
Could you add the line `if type(min)=='number' and type(max)=='number' then` immediately after the `local min,max` line and add another `end` before the last end in that section, so the code reads:
```lua
      -- refresh MIDI controller since mapping has changed
      for _,param in ipairs(Parameters.Order) do
        local min,max = Limits.GetMinMax(param)
        if type(min)=='number' and type(max)=='number' then
          local midivalue = (LrDevelopController.getValue(param)-min)/(max-min) * MIDI2LR.CONTROL_MAX
          MIDI2LR.SERVER:send(string.format('%s %g\n', param, midivalue))
        end
      end
```
I'll get a fix out quickly on this. I added the code to fix one bug but introduced another. :frowning:

**Comment by krakozavr** [comment]
I did it, and I think I did it properly :) (probably not...)

But after changes, MIDI2LR stop working. When I run it directly, it started, but reported as "not connected to MIDI device", and the device make no changes in LR, although MIDI2LR show values changes when I move sliders. When I start it via plugin manager, it swithing to "disabled".
Diagnostic message from Plugin Manager is attached.

[diag.txt](https://github.com/rsjaffe/MIDI2LR/files/86676/diag.txt)

**Comment by rsjaffe** [comment]
OK, new release available to try--doesn't crash on my machine, should have addressed the issue. Turns out that, while I had tested the release, I hadn't tried every possible scenario, and if the profile is changed when in a module other than Develop, I should avoid trying to sync the MIDI controller, as Lightroom can't provide the needed information.

Long story, but hope the release works. Your feedback is appreciated! https://github.com/rsjaffe/MIDI2LR/releases/tag/v0.9.2
